src/pybind/mgr/dashboard/controllers/auth.py

```diff
@@ -186,7 +227,22 @@ def check(self, token):
                     'sso': mgr.SSO_DB.protocol == 'saml2',
                     'pwdUpdateRequired': user.pwd_update_required
                 }
-        return {
-            'login_url': self._get_login_url(),
-            'cluster_status': ClusterModel.from_db().dict()['status']
-        }
+        elif hasattr(cherrypy.request, 'cookie') and '_oauth2_proxy' in cherrypy.request.cookie and \
+          cherrypy.request.cookie['_oauth2_proxy'].value:
```

Comment:
Please, remember to refactor this check on a function such as `is_oauth2_request(cherrypy.request)`

Reply:
refactored it